### DISCUSSION TITLE
Adapting ViSP Example for Robotiq Wrist Camera

### DISCUSSION THREAD

**Discussion by aymane1363** [lagadic/visp#1305] [Q&A]
Hi,

I'm working on implementing Image-Based Visual Servoing with ViSP in Universal Robot UR3e, using the example designed for RealSense cameras provided in the VISP documentation. is it possible transition from a RealSense camera to the Robotiq Wrist Camera in the context of ViSP-based visual servoing?

**Comment by fspindle** [comment]
Of course, the only thing that needs to be done is to write a wrapper on the camera SDK to be able to retrieve the images in the form of `vpImage` (ViSP image format) or `cv::Mat` (OpenCV image format). In the latter case, we provide the `vpImageConvert` class, which converts an OpenCV image into a ViSP image. Once this has been done, you can use the Robotiq Wrist Camera in exactly the same way as a RealSense camera.

**Reply by ER-ROUGUI** [reply]
Hello , thank you 

The wrist camera doesn't have an SDK file, as mentioned by the Robotiq support team:

<<< Hi ,
 
Unfortunately, we do not have such tools available for distribution as we only support the usage of our wrist camera through its Robotiq URCap on UR robots.
 
As a side note, I do know that our wrist camera is detected as a standard webcam when connected to a PC, so you may be able to stream its video feed as one for your project.
 
Best regards,>>>

However, I have been able to access the wrist camera using just the IP address on my PC from this URL: http://<my_robot_ip_address>:4242/current.jpg?annotations=<on> So I have used OpenCV (python) to access the live camera feed. 

How do you think I can adapt the ViSP code to achieve the best results without encountering problems? Thank you

**Comment by fspindle** [comment]
If with OpenCV python you can get the images, it should be possible with OpenCV C++ to get the images as a `cv::Mat`.
If so then use [vpImage::convert()](https://visp-doc.inria.fr/doxygen/visp-daily/classvpImageConvert.html#a099c7f14ecb0549908a656bcb76b3f2a) to convert to a ViSP image.
